zonemaster/zonemaster-engine
# Remove some unintended network dependencies from the unit tests

## Purpose

This PR makes the unit tests don't autocomplete fake glue by sending out DNS requests.

## Context

Fixes #888.

## Changes

A new method Zonemaster::Engine->add_fake_delegation_raw() is factored out from Zonemaster::Engine->add_fake_delegation(). The new method performs the meat of things but without the bells and whistles like autocompleting fake glue and returning a boolean. (I don't think the return value is ever used and it's kind of nonsense anyway. Maybe I should have removed it altogether.)

## How to test this PR

1. In one terminal run `sudo tcpdump -nn port 53` and keep it open.
2. In another terminal run `make test` and wait for it to complete.
3. Make sure tcpdump didn't list any DNS requests from the tests.
